src/AuthContext.tsx

```diff
@@ -192,6 +192,40 @@ export const AuthProvider = ({ authConfig, children }: IAuthProvider) => {
     )
   }
 
+  // This ref is used to make sure the 'fetchTokens' call is only made once.
+  // Multiple calls with the same code will, and should, return an error from the API
+  // See: https://beta.reactjs.org/learn/synchronizing-with-effects#how-to-handle-the-effect-firing-twice-in-development
+  const didFetchTokens = useRef(false)
+
+  function doFetchTokens(urlParams: URLSearchParams): void {
```

Comment:
Could just be called `fetchTokens`

Reply:
> Could just be called `fetchTokens`

There is `fetchTokens` already...

Reply:
Fair point, so this is a wrapper of the fetching, which does more then. Hm.

`performTokenFetching`? 
`handleTokenFetch`? 
Something like that maybe. 

PS: Still waiting on @soofstad's feedback on the idea of the PR